exporter/sumologicexporter/exporter.go

```diff
@@ -85,12 +102,103 @@ func newLogsExporter(
 		se.pushLogsData,
 		// Disable exporterhelper Timeout, since we are using a custom mechanism
 		// within exporter itself
+		exporterhelper.WithTimeout(exporterhelper.TimeoutSettings{Timeout: 0}),
 		exporterhelper.WithRetry(cfg.RetrySettings),
 		exporterhelper.WithQueue(cfg.QueueSettings),
 	)
 }
 
-// pushLogsData groups data with common metadata and send them together to Sumo Logic
-func (se *sumologicexporter) pushLogsData(context.Context, pdata.Logs) (droppedTimeSeries int, err error) {
+// pushLogsData groups data with common metadata and sends them as separate batched requests.
+// It returns the number of unsent logs and error which contains list of dropped records
+// so they can be handled by OTC retry mechanism
+func (se *sumologicexporter) pushLogsData(ctx context.Context, ld pdata.Logs) (int, error) {
+	var (
+		currentMetadata  fields
+		previousMetadata fields
+		errors           []error
+		droppedRecords   []pdata.LogRecord
+		err              error
+	)
+
+	c, err := newCompressor(se.config.CompressEncoding)
+	if err != nil {
+		return 0, consumererror.PartialLogsError(fmt.Errorf("failed to initialize compressor: %w", err), ld)
```

Comment:
Same for the error wrapping. https://godoc.org/github.com/pkg/errors#Wrap can be used for this (retains error metadata not only the formatted string)